pkg/apiserver/handler_impersonation.go

```diff
@@ -92,8 +91,11 @@ func WithImpersonation(handler http.Handler, requestContextMapper api.RequestCon
 				actingAsAttributes.Resource = "groups"
 				groups = append(groups, impersonationRequest.Name)
 
+			case authenticationapi.Kind("UserExtra"):
```

Comment:
Since these are evaluated one at a time, it might be good to document that extra values should be independent of Group and User information. Though I'm trying to think through the implications of that.